scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -358,30 +366,31 @@ class FossId internal constructor(
         revision: String,
         projectCode: String,
         projectName: String
-    ): String {
+    ): Pair<String, String> {
         val existingScan = scans.recentScansForRepository(url, revision).findLatestPendingOrFinishedScan()
 
-        val scanCode = if (existingScan == null) {
+        val scanCodeAndIdPair = if (existingScan == null) {
             logger.info { "No scan found for $url and revision $revision. Creating scan..." }
 
             val scanCode = namingProvider.createScanCode(projectName)
             val newUrl = urlProvider.getUrl(url)
-            createScan(projectCode, scanCode, newUrl, revision)
+            val scanId = createScan(projectCode, scanCode, newUrl, revision)
 
             logger.info { "Initiating the download..." }
             service.downloadFromGit(config.user, config.apiKey, scanCode)
                 .checkResponse("download data from Git", false)
 
-            scanCode
+            scanCode to scanId
         } else {
             logger.info { "Scan '${existingScan.code}' found for $url and revision $revision." }
 
             requireNotNull(existingScan.code) {
                 "The code for an existing scan must not be null."
-            }
+            } to existingScan.id.toString()
```

Comment:
Having an extra line like `existingScan.code to existingScan.id.toString()` would probably be a bit clearer.

Reply:
Likewise, this comment was not addressed (in this commit).

Reply:
@sschuberth wrong commit, I have moved the changes to the respective commit now.